controllers/player.controller.js

```diff
@@ -1,43 +1,90 @@
 const Player = require('../models/player.model')
 
+const logError = requestId => {
+  console.log(
+    JSON.stringify({
+      kinto_request_id: requestId,
+      error: error
+    })
+  )
+}
+
 exports.hello = (req, res) => {
-  res.send('Hello world')
+  const name = req.params.name
+  res.send(`Hello ${name}`)
 }
 
-exports.player_create = (req, res) => {
-  let player = new Player({
-    name: req.body.name,
-    score: req.body.score
-  })
+exports.player_create = async (req, res) => {
+  try {
+    let player
+    player = await Player.find({ name: req.body.name })
 
-  player.save(function(err) {
-    if (err) {
-      return next(err)
+    if (!player.length) {
+      player = new Player({
+        name: req.body.name,
+        score: req.body.score
+      })
     }
-    res.send('Player Created successfully')
-  })
+    await player.save()
+    res.status(200).send(player)
+  } catch (error) {
+    const requestId = req.get('kinto-request-id')
+    logError(requestId)
+    res.send({ error: `player not saved: ${error}` })
+  }
+}
+
+exports.player_details = async (req, res) => {
+  try {
+    let player = await Player.find({ name: req.params.username })
+    const foundPlayer = player[0]
+    player = await Player.findById(foundPlayer._id)
+    res.status(200).send(player)
+  } catch (error) {
+    res.send({ error: `player not found: ${error}` })
+  }
 }
 
-exports.player_details = (req, res) => {
-  Player.findById(req.params.id, (err, player) => {
-    if (err) return next(err)
-    res.send(player)
-  })
+exports.player_update = async (req, res) => {
+  try {
+    let player = await Player.find({ name: req.params.username })
+    const foundPlayer = player[0]
+    const newScore = foundPlayer.score + 1
+    player = await Player.findByIdAndUpdate(
```

Comment:
don't overwrite old variables just create new ones and in this example you don't need to do `findByIdAndUpdate` just update the value and call `.save` after without doing another find